livepeer/catalyst
# Add USER_NEW trigger handler, split handlers code

This PR adds a Mist triggers handler for the catalyst-node at `/triggers` endpoint it also intercepts `USER_NEW` trigger and parse its payload. No logic to control whether to allow or deny the user access to playback.

Refactor catalyst-node to extract endpoints to their own module.

How to test:
* Build the docker image with
```
make docker
```
* Run the following command to launch MistController:
```
docker run \
  --shm-size=1874m  --rm -it \
  -v $(pwd)/config:/config:rw \
  --hostname catalyst-one --name catalyst-one --network catalyst-nodes \
  -p 1935:1935 -p 8080:8080 -p 4242:4242 -p 7373:7373 \
  livepeer/catalyst:latest MistController -c /config/single-catalyst.json
```
* Send a stream to catalyst
```
ffmpeg -re -f lavfi -i testsrc=size=480x320:rate=30 -f lavfi -i sine -vf drawtext="text='%{localtime\:%X}':fontsize=20:fontcolor=white:x=7:y=7" -pix_fmt yuv420p -vcodec libx264 -vb 2000k -preset ultrafast -acodec aac -f flv 'rtmp://127.0.0.1:1935/live/stream+foo'
```

* Try to playback the stream on `http://localhost:4242/`
Note that we are using `config/single-catalyst.json` since it contains the `USER_NEW` trigger config.

Successful handling of the trigger will result in a working playback if `cmd/catalyst-node/handlers/triggers-handler.go`s `handleUserNew` returns `[]byte("true")` or no playback if `[]byte("false")` is returned.

Closes #165
